async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -86,11 +86,11 @@ object AsyncHttpClient {
 
           subscribeF = F.delay(publisher.subscribe(subscriber))
           bodyDisposal <- Ref.of[F, F[Unit]] {
-            subscribeF >> subscriber.stream.take(0).compile.drain
+            subscriber.stream(subscribeF).take(0).compile.drain
           }
 
-          body = Stream.eval(F.uncancelable(bodyDisposal.set(F.unit) >> subscribeF)) >>
-            subscriber.stream.flatMap(part => chunk(Chunk.bytes(part.getBodyPartBytes)))
+          body = Stream.eval(F.uncancelable(bodyDisposal.set(F.unit))) >>
+            subscriber.stream(subscribeF).flatMap(part => chunk(Chunk.bytes(part.getBodyPartBytes)))
```

Comment:
```suggestion
          body = subscriber.stream(
            bodyDisposal.set(F.unit) >> subscribeF).flatMap(part => chunk(Chunk.bytes(part.getBodyPartBytes)))
```

You need the whole thing to be handled atomically, rather than just the subscription. Otherwise, looks good!